src/arch/arm/kernel/boot.c

```diff
@@ -41,19 +41,26 @@ BOOT_BSS static volatile int node_boot_lock;
 
 #define ARCH_RESERVED 3 // kernel + user image + dtb
 #define MAX_RESERVED (ARCH_RESERVED + MODE_RESERVED)
-BOOT_BSS static region_t reserved[MAX_RESERVED];
+BOOT_BSS static region_t res_reg[MAX_RESERVED];
 
-BOOT_CODE static void arch_init_freemem(p_region_t ui_p_reg, region_t dtb_reg, v_region_t it_v_reg,
+BOOT_CODE static void arch_init_freemem(p_region_t ui_p_reg, region_t dtb_reg,
+                                        v_region_t it_v_reg,
                                         word_t extra_bi_size_bits)
 {
-    reserved[0].start = KERNEL_ELF_BASE;
-    reserved[0].end = (pptr_t)ki_end;
+    unsigned int index = 0;
```

Comment:
what principles are you applying to rewrite this code? What's the point here?

Reply:
This is supposed to make it more easy to understand an more compact. And it aligns things between ARM and RISC-V. This patches are from a bigger patch set that I'm upstreaming in smaller pieces. I can drop the last commit from this PR and then include it in a later PR, where this function disappears completely

Reply:
Is the intent to unify code between RISC-V and ARM and then remove this duplication after unifying? If so that makes sense, and I appreciate the effort and the upstreaming. 

Context is really important to help reviewers understand what is going on and will help the review process go smoothly. It would help if that was in the commit messages, because as it stands the comment just says 'clean up the code' which looks at face value like it's being rewritten for no good reason. Note that this policy is described in the guide for pull requests: https://docs.sel4.systems/processes/code-review.html.

I'm very against any unnecessary code churn (which without context, looks like what this is), because:
    - it takes up reviewer time
    - especially when it appears that most of the hardware tests are currently broken (so we don't know the changes don't introduce bugs) *and* its in unverified code (so we *really* don't know).

Reply:
I see you point and have remove the last commit from this PR.

Reply:
> This is supposed to make it more easy to understand an more compact. And it aligns things between ARM and RISC-V. This patches are from a bigger patch set that I'm upstreaming in smaller pieces. I can drop the last commit from this PR and then include it in a later PR, where this function disappears completely

Can you split this single commit into the part that changes Arm and the part that updates RISC-V to be the same as Arm.  The motivations for changing the Arm implementation are different from the motivations to make RISC-V the same as Arm and doing it all in the same change makes it harder to follow what changes are new for Arm and what changes are new for RISC-V

Reply:
I've split up the commits and remove the parameter name change.